hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/HoodieOptionConfig.scala

```diff
@@ -43,6 +43,11 @@ object HoodieOptionConfig {
    */
   val SQL_VALUE_TABLE_TYPE_MOR = "mor"
 
+  /**
+   * The short name for the value of index type.
+   */
+  val SQL_VALUE_INDEX_TYPE = "index.type"
```

Comment:
Maybe we can fix the options in hudi catalog.

Reply:
This issue exists in various Spark versions, such as 2.4, 3.0, etc., while org. apache. park. SQL. hudi. catalog The implementation of HoodieCatalog is only available at 3.2 and above. My current modification is in the configuration of merging Hoodie CatalogTable catalog properties and hoodie. properties records, so I think it is acceptable.

Reply:
`index.type` is a short-cut name only for Flink now.

Reply:
I have referred to your suggestions for modification.When creating a table in Flink, convert index.type to hoodie.index.type